File: text-ui/src/main/java/be/witspirit/mathex/textui/Stats.java
package be.witspirit.mathex.textui;

/**
 * Keeps track of statistics of an exercise session
 */
public class Stats {
    private int total = 0;
    private int correct = 0;
    private int fault = 0;

    /**
     * @return Nr of exercises presented to the user. Excludes repeated sums due to faults
     */
    public int getTotal() {
        return total;
    }

    /**
     * @return Nr of exercises answered correctly at the first try
     */
    public int getCorrect() {
        return correct;
    }

    /**
     * @return Nr of exercises answered faulty. Counts each faulty attempt !
     */
    public int getFault() {
        return fault;
    }

    public Stats incTotal() {
        total++;
        return this;
    }

    public Stats incCorrect() {
        correct++;
        return this;
    }

    public Stats incFault() {
        fault++;
        return this;
    }
}

File: polymer-web-ui/src/main/webapp/resources/js/stats.js
var stats = function() {
    var total = 0;
    var correct = 0;
    var mistakes = 0;

    var reset = function() {
        total = 0;
        correct = 0;
        mistakes = 0;
    };

    var trackCorrect = function() {
        total++;
        correct++;
    };

    var trackMistake = function() {
        total++;
        mistakes++;
    };

    var computeScore = function() {
        return (correct / total) * 100;
    };

    return {
        getTotal : function() { return total; },
        getCorrect : function() { return correct; },
        getMistakes : function() { return mistakes; },
        getScore : computeScore,
        trackCorrect : trackCorrect,
        trackMistake : trackMistake,
        reset : reset
    };
}();
